bioptim/optimization/optimal_control_program.py

```diff
@@ -712,52 +712,69 @@ def load(file_path: str) -> list:
             out = [ocp, sol]
         return out
 
+    def get_dynamics(self):
+        list_dynamics = []
+        for nlp in self.nlp:
+            list_dynamics.append(nlp.dynamics_type.type.name)
+        return list_dynamics
+
+    def get_ode_solver(self):
+        list_ode = []
+        for nlp in self.nlp:
+            list_ode.append(nlp.ode_solver.rk_integrator.__name__)
+        return list_ode
+
     def structure_graph(
         self,
-        n_shooting: Union[int, tuple],
         print_to_terminal: bool = False,
+        draw_graph: bool = False,
     ):
 
-        if isinstance(n_shooting, tuple):
-            n_nodes = sum(n_shooting) + 1
-        else:
-            n_nodes = n_shooting + 1
-
-        init_dict = [["Phase", []], ["Mayer", []], ["Lagrange", []], ["Constraints", []]]
-        init_dict = dict(init_dict)
-        list_nodes = [init_dict] * n_nodes
-        for i in range(n_nodes):
-            list_nodes[i] = deepcopy(init_dict)
+        def merge_dicts(a, b):
+            m = a.copy()
+            m.update(b)
+            return m
+
+        def print_terminal(print_terminal: bool, l_dynamics: list, l_ode: list, l_nodes: list, n_phase: int):
+            if print_terminal is True:
```

Comment:
Put the if ouside of the function, if the function is called, there is no reason for it to doubt that it should do its job. This is the caller's job to know if it wants or not something

Reply:
Done